### ISSUE TITLE
It does not give a warning when the return type is determined in the Protocol class and the type is not specified in the subclass.

### ISSUE THREAD

**Issue by ozkansen** [python/mypy#11091]
It does not give a warning when the return type is determined in the Protocol class and the type is not specified in the subclass.

```python
from typing import Protocol


class Main(Protocol):
    def validate(self) -> str:
        ...


class Sub:
    def validate(self):
        ...


def run(sub: Main):
    sub.validate()


sub_class = Sub()
run(sub_class)
```

When I check with mypy it finds no problem, but here the return type is set to str.

```
Success: no issues found in 1 source file
```

**Comment by JelleZijlstra** [comment]
Mypy generally doesn't typecheck unannotated methods. You can use `--disallow-untyped-defs` to get an error here.

**Comment by ozkansen** [comment]
yes i was able to get the following error with this method but this is different from what i expected

```bash
$ mypy --disallow-untyped-defs main.py

main.py:10: error: Function is missing a return type annotation
main.py:10: note: Use "-> None" if function does not return a value
main.py:14: error: Function is missing a return type annotation
Found 2 errors in 1 file (checked 1 source file)
```

-----
When I edit the return value to None I get the following error, isn't that what it should be?

```python
class Sub:
    def validate(self) -> None:
        ...
```

```bash
$ mypy --disallow-untyped-defs main.py

main.py:14: error: Function is missing a return type annotation
main.py:19: error: Argument 1 to "run" has incompatible type "Sub"; expected "Main"
main.py:19: note: Following member(s) of "Sub" have conflicts:
main.py:19: note:     Expected:
main.py:19: note:         def validate(self) -> str
main.py:19: note:     Got:
main.py:19: note:         def validate(self) -> None
Found 2 errors in 1 file (checked 1 source file)
```

**Comment by ozkansen** [comment]
> Mypy generally doesn't typecheck unannotated methods. You can use `--disallow-untyped-defs` to get an error here.

Yeah, it is normal for me to experience the above situation if mypy is not checking without specifying the value.

**Comment by JelleZijlstra** [comment]
If you don't specify a return annotation, mypy assumes `-> Any`, and that's compatible with `-> str`.

**Comment by ozkansen** [comment]
i see thank you
